crates/sui-core/src/consensus_adapter.rs

```diff
@@ -187,6 +192,33 @@ impl ConsensusAdapterMetrics {
     }
 }
 
+pub enum BlockStatus {
+    Sequenced,
+    GarbageCollected,
+}
```

Comment:
In theory we could just use directly the Mysticeti block status type, but I've tried to avoid as much as possible at the moment to leak the types on the Sui side. Maybe not that necessary - let me know if you think otherwise.